src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,146 @@
+                    @Override
+                    public void onEvent(long id, String eventType, Bundle eventData) {
+                    }
+
+                    @Override
+                    public void onDone(long lastId) {
+                        // Now we know the last event ID. We check if it ran backwards and then start loading events.
+                        if (lastId < mLastEventId) mLastEventId = 0;
+
+                        if (Log.isLoggable(TAG, Log.DEBUG)) Log.d(TAG, "Reading events starting with id " + mLastEventId);
+
+                        mApi.getEvents(mLastEventId, 0, new RestApi.OnReceiveEventListener() {
+                            @Override
+                            public void onEvent(final long id, final String eventType, final Bundle eventData) {
+                                // If a folder item is contained within the event. Resolve the local path.
+                                if (eventData.containsKey("folder")) {
+                                    String folderPath = null;
+                                    synchronized (mFolderToPath) {
+                                        if (mFolderToPath.size() == 0) updateFolderMap();
+                                        folderPath = mFolderToPath.get(eventData.getString("folder"));
+                                    }
+
+                                    if (folderPath != null) {
+                                        eventData.putString("_localFolderPath",folderPath);
+
+                                        if (eventData.containsKey("item")) {
+                                            final File file = new File(new File(folderPath), eventData.getString("item"));
+
+                                            eventData.putString("_localItemPath", file.getPath());
+                                        }
+                                    }
+                                }
+
+                                final Intent eventBroadcastItent = new Intent(EVENT_BASE_URL + "." + eventType.toUpperCase());
+                                eventBroadcastItent.putExtras(eventData);
+                                mLocalBM.sendBroadcast(eventBroadcastItent);
+
+                                if (Log.isLoggable(TAG, Log.DEBUG)) Log.d(TAG, "Sent local event broadcast " + eventBroadcastItent.getAction() + " including " + eventType.length() + " extra data items.");
+                            }
+
+                            @Override
+                            public void onDone(long id) {
+                                if (mLastEventId < id) {
+                                    mLastEventId = id;
+
+                                    // Store the last EventId in case we get killed
+                                    final SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(mContext);
+                                    sp.edit().putLong(PREF_LASTSYNCID, mLastEventId).commit();
```

Comment:
Ah that makes sense. But you should disable the warning then, with `//noinspection CommitPrefEdits`.